crates/forge_analyzer/src/checkers.rs

```diff
@@ -926,61 +927,60 @@ impl<'cx> Runner<'cx> for SecretChecker {
-                                interp.get_value(def, varid, Some(var.projections.clone()))
-                            {
-                                match value {
-                                    Value::Const(_) | Value::Phi(_) => {
-                                        let vuln = SecretVuln::new(
-                                            interp.callstack(),
-                                            interp.env(),
-                                            interp.entry(),
-                                        );
-                                        info!("Found a vuln!");
-                                        self.vulns.push(vuln);
-                                    }
-                                    _ => {}
-                                }
-                            } else if let Some(VarKind::GlobalRef(def)) =
-                                interp.body().vars.get(varid)
-                            {
-                                if let Some(value) = interp.value_manager.defid_to_value.get(def) {
-                                    println!("value [] {value:?}");
-                                    match value {
-                                        Value::Const(_) | Value::Phi(_) => {
-                                            let vuln = SecretVuln::new(
-                                                interp.callstack(),
-                                                interp.env(),
-                                                interp.entry(),
-                                            );
-                                            info!("Found a vuln!");
-                                            self.vulns.push(vuln);
-                                        }
-                                        _ => {}
-                                    }
-                                }
-                            }
+                }
+            }
+            Intrinsic::Fetch => {
+                if let Some(Operand::Var(Variable {
+                    base: Base::Var(varid),
+                    ..
+                })) = operands.unwrap_or_default().get(1)
+                {
+                    let varid_argument =
+                        if let Some(Value::Object(varid)) = interp.get_value(def, *varid, None) {
+                            varid
+                        } else {
+                            varid
+                        };
+                    let headers_proj = projvec_from_str("Headers");
+                    let headers_proj_lower = projvec_from_str("headers");
+                    if let Some(Value::Object(varid)) = interp
+                        .get_value(def, *varid_argument, Some(headers_proj))
+                        .or_else(|| {
+                            interp.get_value(def, *varid_argument, Some(headers_proj_lower))
+                        })
+                    {
+                        let auth_proj = projvec_from_str("Authorization");
```

Comment:
Technically, any uppercase/lowercase combination of "authorization" works, however, these two are likely to be the only ones people use, so handling all cases can be a separate PR/task unless you want to do it in this one.

Reply:
should we loop through all of the characters, or do you think there is an alternative way to do this? Maybe make any vecs with properties authorization all lower case?